Instant Search : Update mobile structure

keoshi opened the pull request:
#### Changes proposed in this Pull Request:

* Use cross instead of label in the button to close overlay (reported in https://github.com/Automattic/jetpack/pull/14379#pullrequestreview-344742333 ).
* Changes position of close overlay button to the corner.
* Fixes z-index stack order so close button is always usable.
* Adds mobile styles for the popover.
* Adds medium break point at `768px`.

![instant-search-mobile-light](https://user-images.githubusercontent.com/390760/73192885-a408fc80-4121-11ea-8b15-f023d0472aea.png)

![instant-search-mobile-dark](https://user-images.githubusercontent.com/390760/73192888-a5d2c000-4121-11ea-962a-8362fc428071.png)


#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:

1. Follow the [testing instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search.
1. Enter search term and see results on overlay.
1. Ensure all elements work.

**Note for devs**: ~I will need some help to toggle visibility of the filters pane / widget area when using a mobile device. Saw some related code, but couldn't get it working. This could mean all popover styles I've added here are misplaced and need relocating to a different class.~ This was added by @jsnmoon in https://github.com/Automattic/jetpack/pull/14451

#### Proposed changelog entry for your changes:

* None needed.

keoshi committed:
Add cross gridicon

keoshi committed:
Replace close button label by a gridicon

keoshi committed:
Change position of close button

keoshi committed:
Update z-index stacking order

Prevents sidebar to be in front of the close button

keoshi committed:
Reorder CSS properties

keoshi committed:
Add medium break point at 768px

keoshi committed:
Add mobile styles for filters popover

keoshi committed:
Change how transparency is set on the overlay

jsnmoon commented on modules/search/instant-search/components/search-results.scss:
```diff
@@ -19,7 +20,27 @@
 }
 .jetpack-instant-search__search-results-secondary {
-	flex: 2;
+	display: none; // TODO: toggle visibility via JS?
```
It looks like this `display: none` is never overridden in `@include break-medium()`?

jsnmoon reviewed (changes requested):
I'm having trouble getting the sidebar to render; I think I found the culprit in `search-results.scss` below.

jsnmoon commented:
> I will need some help to toggle visibility of the filters pane / widget area when using a mobile device. Saw some related code, but couldn't get it working. 

Our existing filter visibility toggle could probably use some refactoring; I'd be happy to spin up a PR following this one.

Would we want to show the entire overlay sidebar inside the mobile popover?

keoshi commented:
@jsnmoon First off, thanks for reviewing.

[I left this line](https://github.com/Automattic/jetpack/pull/14423/files#diff-dfffc551cadddd6c918afed41a505bcdR23) on purpose so we wouldn't forget to address it, but also because I don't know if these styles make sense in `.jetpack-instant-search__search-results-secondary`.

I'd be inclined to leave this PR on hold until the toggle is in place. That way I can then adjust styles in this PR to match the final mobile popover; what do you think?

> Would we want to show the entire overlay sidebar inside the mobile popover?

I'd say it only makes sense to show the filters here, since the icon to trigger it sets that expectation. Whatever additional widgets customers add to the sidebar would not be included. We'd still have to figure out what to do with them. The two options that come to mind right now are:

- Hide them in mobile (not really a great option since one would expect them to always work).
- Put them at the end of the page after search results (unsure if this makes sense or is usable).

Does this separation between what are filters and what are purely widgets complicate the logic for the popover?

jsnmoon commented:
> I'd be inclined to leave this PR on hold until the toggle is in place. That way I can then adjust styles in this PR to match the final mobile popover; what do you think?

👍 Makes a lot of sense. 

> I'd say it only makes sense to show the filters here, since the icon to trigger it sets that expectation. Whatever additional widgets customers add to the sidebar would not be included. 

Agreed, showing only the filters in the popover makes the most sense. 

>We'd still have to figure out what to do with them. The two options that come to mind right now are:
>
>- Hide them in mobile (not really a great option since one would expect them to always work).
>- Put them at the end of the page after search results (unsure if this makes sense or is usable).

Putting them after the search results at the bottom make a lot of sense. 

>Does this separation between what are filters and what are purely widgets complicate the logic for the popover?

Yes, but I think we can make it work 😄

keoshi committed:
Merge branch 'instant-search-master' into update/instant-search-mobile-structure

keoshi committed:
Update break-small to break-medium

keoshi committed:
Add styles for smaller close button on mobile devices

keoshi committed:
Add arrow element to popover

keoshi committed:
Remove styles that are out of place

Given the markup has changed in https://github.com/Automattic/jetpack/pull/14477, these styles need to be applied elsewhere

keoshi committed:
Update search results styling

keoshi committed:
Add popover styles

keoshi committed:
Update filter icon margins

keoshi commented:
@jsnmoon @bluefuton @gibrown This PR is ready for another round of reviews. Thanks in advance!

bluefuton commented:
This is looking great @keoshi 🎉

One tiny suggestion: shall we remove the `margin-left` from the sort order select and add a right margin to the label instead? It currently means the select is slightly indented when it wraps:

<img width="265" alt="Screen Shot 2020-01-28 at 11 40 46" src="https://user-images.githubusercontent.com/17325/73220424-3f9e7980-41c3-11ea-9875-8de5c77067c0.png">

jsnmoon reviewed (approved):
This is great. 🚢🚢🚢

jsnmoon commented on modules/search/instant-search/components/search-sidebar.scss:
```diff
@@ -1,8 +1,7 @@
 	.jetpack-instant-search-wrapper {
-		// TODO: To be updated to break-medium
-		@media ( max-width: #{ ( $break-small ) } ) {
+		@media ( max-width: #{ ( $break-medium ) } ) {
```
We might actually need the following:
```suggestion
		@media ( max-width: #{ ( $break-medium + 1 ) } ) {
```

keoshi replied:
Great catch! I just noticed by browser window resizes on every two pixels, so it skipped this exact resolution. Fixed in https://github.com/Automattic/jetpack/pull/14423/commits/f506922bed3d370443e3c14bef73f4d3e54071cb

jsnmoon reviewed (changes requested):
Sorry, it looks like my review was premature.

When the screen width is set exactly to 768 pixels, interacting with filters contlos become impossible; sidebar filters and filter toggles are both hidden at this width.

Since iPads use this viewport, we should probably fix this issue before shipping.

keoshi committed:
Show sidebar filters up to 768px

See: https://github.com/Automattic/jetpack/pull/14423#pullrequestreview-349057505

keoshi committed:
Bring up the opacity on the arrow border

keoshi committed:
Reduce side padding on small screens

keoshi committed:
Move margin to label span instead of select element

keoshi commented:
@bluefuton Moved the margin to a new `span` element in https://github.com/Automattic/jetpack/pull/14423/commits/663484e689cf1cae4e4d972b1a63e5e01963f91a and also reduced the overall side padding on https://github.com/Automattic/jetpack/pull/14423/commits/7edd00810501eafda36cc94d9a3b40fa6702cda2 to help maintain the original structure for longer.

keoshi commented:
@bluefuton @jsnmoon Thanks a bunch for your reviews. Pushed a couple more commits and I think this is ready for a final review and merge! 🤞

gibrown commented:
This is looking great.

jsnmoon reviewed (approved):
Looks solid!
